manim/mobject/numbers.py

```diff
@@ -189,13 +191,8 @@ def set_value(self, number, **config):
         new_decimal.scale(self[-1].height / new_decimal[-1].height)
         new_decimal.move_to(self, self.edge_to_fix)
         new_decimal.match_style(self)
+        self.become(new_decimal)
 
-        old_family = self.get_family()
-        self.submobjects = new_decimal.submobjects
-        for mob in old_family:
-            # Dumb hack...due to how scene handles families
-            # of animated mobjects
-            mob.points[:] = 0
```

Comment:
Is this change actually safe?

Reply:
I'm not 100% sure. I tried it with few scenes, and it worked fine. I have no idea why this hack is necessary, but the logic now seems fine to me. It would be nice to have other people's opinion about this change and make sure it is safe.